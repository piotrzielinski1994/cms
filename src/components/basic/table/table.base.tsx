import { HtmlProps } from '@/utils/html/html.types';
import { forwardRef } from 'react';

const Root = (props: HtmlProps['div']) => <div {...props} />;

const Native = forwardRef<HTMLTableElement, HtmlProps['table']>((props, ref) => {
  return <table ref={ref} {...props} />;
});

const Header = (props: HtmlProps['thead']) => <thead {...props} />;
const Body = (props: HtmlProps['tbody']) => <tbody {...props} />;
const Footer = (props: HtmlProps['tfoot']) => <tfoot {...props} />;
const Row = (props: HtmlProps['tr']) => <tr {...props} />;
const Column = (props: HtmlProps['td']) => <td {...props} />;

Native.displayName = 'TableBase.Native';

const TableBase = {
  Root,
  Native,
  Header,
  Body,
  Footer,
  Row,
  Column,
};

export default TableBase;
