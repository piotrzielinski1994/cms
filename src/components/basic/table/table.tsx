import { cn } from '@/utils/tailwind';
import { HTMLAttributes, ReactNode } from 'react';

type TableProps = HTMLAttributes<HTMLTableElement> & {
  header: ReactNode[];
  body: ReactNode[][];
  footer?: ReactNode[];
};

// We always map through header columns in case body or footer are of different length
// Static tuple declarations won't work as we always .map before passing props -> loosing tuple
const Table = ({ header, body, footer = [], ...props }: TableProps) => {
  const hasEvenElements = body.length % 2 === 0;
  return (
    <div className="w-full overflow-x-auto">
      <table {...props} className={cn('w-full', props.className)}>
        <thead className="font-semibold">
          <tr className="bg-primary text-primary-foreground">
            {header.map((cell, index) => {
              return (
                <td key={index} className={cn('px-4 py-2 md:px-6 md:py-4')}>
                  {cell}
                </td>
              );
            })}
          </tr>
        </thead>
        <tbody>
          {body.map((row, index) => {
            const isEven = index % 2 === 0;
            const cells = header.map((_, index) => {
              const cell = row[index];
              return (
                <td key={index} className="px-4 py-2 md:px-6 md:py-4">
                  {cell}
                </td>
              );
            });
            return (
              <tr
                key={index}
                className={cn({
                  'bg-background': isEven,
                  'bg-background1': !isEven,
                })}
              >
                {cells}
              </tr>
            );
          })}
        </tbody>
        {footer.length > 0 && (
          <tfoot className="font-semibold">
            <tr
              className={cn({
                'bg-background': hasEvenElements,
                'bg-background1': !hasEvenElements,
              })}
            >
              {header.map((_, index) => {
                const cell = footer[index];
                return (
                  <td key={index} className="px-4 py-2 md:px-6 md:py-4">
                    {cell}
                  </td>
                );
              })}
            </tr>
          </tfoot>
        )}
      </table>
    </div>
  );
};

export { Table };
