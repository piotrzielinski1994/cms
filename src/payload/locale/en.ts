const en = {
  common: {
    content: 'Content',
    layout: 'Layout',
    component: {
      singular: 'Component',
      plural: 'Components',
    },
    section: {
      singular: 'Section',
      plural: 'Sections',
    },
    related: {
      singular: 'Related',
      plural: 'Related',
    },
    openInNewTab: 'Open in new tab',
    item: {
      singular: 'Item',
      plural: 'Items',
    },
    files: {
      singular: 'File',
      plural: 'Files',
    },
    default: 'Default',
    dark: 'Dark',
    question: 'Question',
    answer: 'Answer',
    pageNotFound: 'This page could not be found.',
    goHome: 'Go home',
  },
  fields: {
    label: 'Label',
    slug: 'Slug',
    path: 'Path',
    type: 'Type',
    title: 'Title',
    heading: 'Heading',
    subheading: 'Subheading',
    description: 'Description',
    image: 'Image',
    publishedAt: 'Published at',
    fullName: 'Full name',
    parent: 'Parent element',
    children: 'Child elements',
    role: {
      singular: 'Role',
      plural: 'Roles',
    },
    menu: 'Menu',
    submenu: 'Submenu',
    navItem: 'Navigation item',
    internalLink: 'Internal link',
    documentToLinkTo: 'Document to link to',
    buttons: 'Buttons',
    selector: 'Selector',
    isReversed: 'Reversed layout',
    subpages: 'Subpages',
    seo: 'SEO',
    acceptLabel: 'Accept label',
    readMoreLabel: 'Read more label',
    url: 'URL',
  },
  groups: {
    admin: 'Admin',
    docs: 'Docs',
  },
  collections: {
    pages: {
      singular: 'Page',
      plural: 'Pages',
    },
    categories: {
      singular: 'Category',
      plural: 'Categories',
    },
    posts: {
      singular: 'Post',
      plural: 'Posts',
    },
    sites: {
      singular: 'Site',
      plural: 'Sites',
    },
    users: {
      singular: 'User',
      plural: 'Users',
    },
    images: {
      singular: 'Image',
      plural: 'Images',
    },
    redirects: {
      singular: 'Redirect',
      plural: 'Redirects',
    },
  },
  enums: {
    role: {
      user: 'User',
      editor: 'Editor',
      admin: 'Admin',
    },
  },
  components: {
    header: 'Header',
    footer: 'Footer',
    cookiesBanner: 'Cookies banner',
  },
  frontend: {
    submit: 'Submit',
    logo: 'Logo',
    localeSwitcher: 'Locale switcher',
    themeSwitcher: 'Theme switcher',
    fontScaleSwitcher: 'Font scale switcher',
    increment: 'Increment',
    decrement: 'Decrement',
    close: 'Close',
    component: {
      skipLink: 'Go to the main content',
    },
    contactForm: {
      fields: {
        email: {
          label: 'Email',
        },
        message: {
          label: 'Message',
        },
      },
    },
  },
  zod: {
    invalid_type: 'Expected {expected}, received {received}',
    invalid_type_with_path: '{path} is expected {expected}, but received {received}',
    invalid_type_received_undefined: 'Required',
    invalid_literal: 'Invalid literal value, expected {expected}',
    unrecognized_keys: 'Unrecognized key(s) in object: {- keys}',
    unrecognized_keys_one: 'Unrecognized key in object: {- keys}',
    unrecognized_keys_other: 'Unrecognized keys in object: {- keys}',
    invalid_union: 'Invalid input',
    invalid_union_discriminator: 'Invalid discriminator value. Expected {- options}',
    invalid_enum_value: 'Invalid enum value. Expected {- options}, received {received}',
    invalid_arguments: 'Invalid function arguments',
    invalid_return_type: 'Invalid function return type',
    invalid_date: 'Invalid date',
    custom: 'Invalid input',
    invalid_intersection_types: 'Intersection results could not be merged',
    not_multiple_of: 'Number must be a multiple of {multipleOf}',
    not_finite: 'Number must be finite',
    invalid_string: {
      email: 'Invalid {validation}',
      url: 'Invalid {validation}',
      uuid: 'Invalid {validation}',
      cuid: 'Invalid {validation}',
      regex: 'Invalid',
      datetime: 'Invalid {validation}',
      startsWith: 'Invalid input: must start with {startsWith}',
      endsWith: 'Invalid input: must end with {endsWith}',
    },
    too_small: {
      array: {
        inclusive: 'Array must contain at least {minimum} element(s)',
        inclusive_one: 'Array must contain at least {minimum} element',
        inclusive_other: 'Array must contain at least {minimum} elements',
        not_inclusive: 'Array must contain more than {minimum} element(s)',
        not_inclusive_one: 'Array must contain more than {minimum} element',
        not_inclusive_other: 'Array must contain more than {minimum} elements',
      },
      string: {
        inclusive: 'String must contain at least {minimum} character(s)',
        inclusive_one: 'String must contain at least {minimum} character',
        inclusive_other: 'String must contain at least {minimum} characters',
        inclusive_with_path: '{path} must contain at least {minimum} character(s)',
        inclusive_with_path_one: '{path} must contain at least {minimum} character',
        inclusive_with_path_other: '{path} must contain at least {minimum} characters',
        not_inclusive: 'String must contain over {minimum} character(s)',
        not_inclusive_one: 'String must contain over {minimum} character',
        not_inclusive_other: 'String must contain over {minimum} characters',
        not_inclusive_with_path: '{path} must contain over {minimum} character(s)',
        not_inclusive_with_path_one: '{path} must contain over {minimum} character',
        not_inclusive_with_path_other: '{path} must contain over {minimum} characters',
      },
      number: {
        inclusive: 'Number must be greater than or equal to {minimum}',
        inclusive_with_path: '{path} must be greater than or equal to {minimum}',
        not_inclusive: 'Number must be greater than {minimum}',
        not_inclusive_with_path: '{path} must be greater than {minimum}',
      },
      set: {
        inclusive: 'Invalid input',
        not_inclusive: 'Invalid input',
      },
      date: {
        inclusive: 'Date must be greater than or equal to {- minimum, datetime}',
        not_inclusive: 'Date must be greater than {- minimum, datetime}',
      },
    },
    too_big: {
      array: {
        inclusive: 'Array must contain at most {maximum} element(s)',
        inclusive_one: 'Array must contain at most {maximum} element',
        inclusive_other: 'Array must contain at most {maximum} elements',
        not_inclusive: 'Array must contain less than {maximum} element(s)',
        not_inclusive_one: 'Array must contain less than {maximum} element',
        not_inclusive_other: 'Array must contain less than {maximum} elements',
      },
      string: {
        inclusive: 'String must contain at most {maximum} character(s)',
        inclusive_one: 'String must contain at most {maximum} character',
        inclusive_other: 'String must contain at most {maximum} characters',
        inclusive_with_path: '{path} must contain at most {maximum} character(s)',
        inclusive_with_path_one: '{path} must contain at most {maximum} character',
        inclusive_with_path_other: '{path} must contain at most {maximum} characters',
        not_inclusive: 'String must contain under {maximum} character(s)',
        not_inclusive_one: 'String must contain under {maximum} character',
        not_inclusive_other: 'String must contain under {maximum} characters',
        not_inclusive_with_path: '{path} must contain under {maximum} character(s)',
        not_inclusive_with_path_one: '{path} must contain under {maximum} character',
        not_inclusive_with_path_other: '{path} must contain under {maximum} characters',
      },
      number: {
        inclusive: 'Number must be less than or equal to {maximum}',
        inclusive_with_path: '{path} must be less than or equal to {maximum}',
        not_inclusive: 'Number must be less than {maximum}',
        not_inclusive_with_path: '{path} must be less than {maximum}',
      },
      set: {
        inclusive: 'Invalid input',
        not_inclusive: 'Invalid input',
      },
      date: {
        inclusive: 'Date must be smaller than or equal to {- maximum, datetime}',
        not_inclusive: 'Date must be smaller than {- maximum, datetime}',
      },
    },
  },
  storybook: {
    basic: {
      form: {
        textInput: {
          default: {
            label: 'Text Input',
            placeholder: 'Type value',
          },
          disabled: {
            label: 'Disabled Input',
            placeholder: 'Disabled',
          },
          invalid: {
            label: 'Invalid Input',
            value: 'Invalid value',
            error: 'Error message',
          },
        },
        numberInput: 'Number Input',
        textArea: 'Text Area',
        checkbox: {
          default: {
            label: 'Checkbox',
          },
          disabled: {
            label: 'Disabled Checkbox',
          },
          invalid: {
            label: 'Invalid Checkbox',
            error: 'Error message',
          },
        },
        radio: {
          default: {
            label: 'Radio Button',
          },
          disabled: {
            label: 'Disabled Radio Button',
          },
          invalid: {
            label: 'Invalid Radio Button',
            error: 'Error message',
          },
        },
        select: {
          label: 'Select',
          placeholder: 'Choose value',
          option: 'Option',
        },
      },
      button: {
        default: 'Button',
        disabled: 'Disabled',
        secondary: 'Secondary',
      },
      alert: {
        info: 'Information',
        success: 'Success',
        warn: 'Warning',
        error: 'Error',
      },
      badge: 'Badge',
      accordion: {
        heading: 'Heading',
        content: 'Content',
      },
      dialog: {
        button: 'Show Dialog',
        header: 'Header',
        content: 'Content',
        submit: 'Submit',
        cancel: 'Cancel',
      },
    },
  },
};

export { en };
