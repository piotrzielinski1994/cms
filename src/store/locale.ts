import { contentLocales, defaultContentLocale } from '@/config/locales.config';
import { Locale, useLocale } from 'next-intl';

const useLocaleStore = (): Locale => {
  const locale = useLocale();
  return contentLocales.includes(locale) ? locale : defaultContentLocale;
};

export { useLocaleStore };
