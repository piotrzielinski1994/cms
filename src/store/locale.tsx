'use client';

import { contentLocales, defaultContentLocale, translations } from '@/config/store/locales.config';
import { Locale, NextIntlClientProvider, useLocale } from 'next-intl';
import { PropsWithChildren, useEffect } from 'react';
import z from 'zod';

const useLocaleStore = (): Locale => {
  const locale = useLocale();
  return contentLocales.includes(locale) ? locale : defaultContentLocale;
};

const LocaleProvider = ({ children, locale }: PropsWithChildren & { locale: Locale }) => {
  useEffect(() => {
    z.config(z.locales[locale]());
  }, [locale]);

  return (
    <NextIntlClientProvider locale={locale} messages={translations[locale]}>
      {children}
    </NextIntlClientProvider>
  );
};

export { LocaleProvider, useLocaleStore };
