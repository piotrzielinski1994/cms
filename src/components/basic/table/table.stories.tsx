import type { Meta, StoryObj } from '@storybook/react';
import { useTranslations } from 'next-intl';
import { type ComponentProps } from 'react';
import { Table as TableComponent } from './table';

type Args = ComponentProps<typeof TableComponent>;

const meta = {
  component: TableComponent,
  title: 'Basic/Table',
  argTypes: {},
  args: {},
} satisfies Meta<Args>;

const Render = ({ ...args }: Args) => {
  const t = useTranslations('storybook.basic.table');
  const [cols, rows] = [3, 5];

  const header = Array.from({ length: cols }, (_, index) => `${t('header')} ${index + 1}`);
  const footer = Array.from({ length: cols }, (_, index) => `${t('footer')} ${index + 1}`);
  const cells = Array.from({ length: rows }, (_, rowIndex) =>
    Array.from({ length: cols }, (_, index) => `${t('cell')} ${rowIndex * cols + index + 1}`),
  );

  return (
    <div className="flex gap-1">
      <TableComponent {...args} header={header} footer={footer} body={cells} />
    </div>
  );
};

const Table: StoryObj<typeof TableComponent> = { render: Render };

export { Table };
export default meta;
