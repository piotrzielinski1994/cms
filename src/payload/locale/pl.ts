import { en } from './en';

const pl = {
  common: {
    content: 'Treść',
    layout: 'Układ',
    component: {
      singular: 'Komponent',
      plural: 'Komponenty',
    },
    section: {
      singular: 'Sekcja',
      plural: 'Sekcje',
    },
    related: {
      singular: 'Powiązany',
      plural: 'Powiązane',
    },
    openInNewTab: 'Otwórz w nowej karcie',
    item: {
      singular: 'Element',
      plural: 'Elementy',
    },
    files: {
      singular: 'Plik',
      plural: 'Pliki',
    },
    default: 'Domyślny',
    dark: 'Ciemny',
    question: 'Pytanie',
    answer: 'Odpowiedź',
    pageNotFound: 'Nie znaleziono strony.',
    goHome: 'Wróć do strony głównej',
  },
  fields: {
    label: 'Etykieta',
    slug: 'Slug',
    path: 'Ścieżka',
    type: 'Typ',
    title: 'Tytuł',
    heading: 'Naglówek',
    subheading: 'Podnagłówek',
    description: 'Opis',
    image: 'Zdjęcie',
    publishedAt: 'Data publikacji',
    fullName: 'Imię i nazwisko',
    parent: 'Element nadrzędny',
    children: 'Elementy podrzędne',
    role: {
      singular: 'Rola',
      plural: 'Role',
    },
    menu: 'Menu',
    submenu: 'Submenu',
    navItem: 'Element nawigacji',
    internalLink: 'Wewnętrzny link',
    documentToLinkTo: 'Dokument do podlinkowania',
    buttons: 'Przyciski',
    selector: 'Selektor',
    isReversed: 'Odwrócony układ',
    subpages: 'Podstrony',
    seo: 'SEO',
    acceptLabel: 'Etykieta akceptacji',
    readMoreLabel: 'Etykieta przeczytaj więcej',
    url: 'URL',
  },
  groups: {
    admin: 'Administrator',
    docs: 'Dokumentacja',
  },
  collections: {
    pages: {
      singular: 'Strona',
      plural: 'Strony',
    },
    categories: {
      singular: 'Kategoria',
      plural: 'Kategorie',
    },
    posts: {
      singular: 'Post',
      plural: 'Posty',
    },
    sites: {
      singular: 'Witryna',
      plural: 'Witryny',
    },
    users: {
      singular: 'Użytkownik',
      plural: 'Użytkownicy',
    },
    images: {
      singular: 'Zdjęcie',
      plural: 'Zdjęcia',
    },
    redirects: {
      singular: 'Przekierowanie',
      plural: 'Przekierowania',
    },
  },
  enums: {
    role: {
      user: 'Użytkownik',
      editor: 'Edytor',
      admin: 'Administrator',
    },
  },
  components: {
    header: 'Nagłówek',
    footer: 'Stopka',
    cookiesBanner: 'Baner z ciasteczkami',
  },
  frontend: {
    submit: 'Zatwierdź',
    logo: 'Logo',
    localeSwitcher: 'Przełącznik języka',
    themeSwitcher: 'Przełącznik motywu',
    fontScaleSwitcher: 'Przełącznik skalowania czcionki',
    increment: 'Zwiększ',
    decrement: 'Zmniejsz',
    component: {
      skipLink: 'Przejdź do głównej zawartości',
    },
    contactForm: {
      fields: {
        email: {
          label: 'Email',
        },
        message: {
          label: 'Wiadomość',
        },
      },
    },
  },
  zod: {
    invalid_type: 'Oczekiwano {expected}, otrzymano {received}',
    invalid_type_with_path: '{path} powinno być typu {expected}, ale jest typu {received}',
    invalid_type_received_undefined: 'Wymagane',
    invalid_literal: 'Nieprawidłowa wartość literalna, oczekiwano {expected}',
    unrecognized_keys: 'Nierozpoznane klucze w obiekcie: {- keys}',
    unrecognized_keys_one: 'Nierozpoznany klucz w obiekcie: {- keys}',
    unrecognized_keys_other: 'Nierozpoznane klucze w obiekcie: {- keys}',
    invalid_union: 'Nieprawidłowe dane wejściowe',
    invalid_union_discriminator:
      'Nieprawidłowa wartość dyskryminatora. Oczekiwano jednej z: {- options}',
    invalid_enum_value:
      'Nieprawidłowa wartość enum. Oczekiwano jednej z: {- options}, otrzymano {received}',
    invalid_arguments: 'Nieprawidłowe argumenty funkcji',
    invalid_return_type: 'Nieprawidłowy typ zwracany przez funkcję',
    invalid_date: 'Nieprawidłowa data',
    custom: 'Nieprawidłowe dane',
    invalid_intersection_types: 'Nie można połączyć wyników przecięcia',
    not_multiple_of: 'Liczba musi być wielokrotnością {multipleOf}',
    not_finite: 'Liczba musi być skończona',
    invalid_string: {
      email: 'Nieprawidłowy {validation}',
      url: 'Nieprawidłowy {validation}',
      uuid: 'Nieprawidłowy {validation}',
      cuid: 'Nieprawidłowy {validation}',
      regex: 'Nieprawidłowy format',
      datetime: 'Nieprawidłowy {validation}',
      startsWith: 'Nieprawidłowa wartość: musi zaczynać się od {startsWith}',
      endsWith: 'Nieprawidłowa wartość: musi kończyć się na {endsWith}',
    },
    too_small: {
      array: {
        inclusive: 'Tablica musi zawierać co najmniej {minimum} element(ów)',
        inclusive_one: 'Tablica musi zawierać co najmniej {minimum} element',
        inclusive_other: 'Tablica musi zawierać co najmniej {minimum} elementy',
        not_inclusive: 'Tablica musi zawierać więcej niż {minimum} element(ów)',
        not_inclusive_one: 'Tablica musi zawierać więcej niż {minimum} element',
        not_inclusive_other: 'Tablica musi zawierać więcej niż {minimum} elementy',
      },
      string: {
        inclusive: 'Ciąg musi zawierać co najmniej {minimum} znak(ów)',
        inclusive_one: 'Ciąg musi zawierać co najmniej {minimum} znak',
        inclusive_other: 'Ciąg musi zawierać co najmniej {minimum} znaki',
        inclusive_with_path: '{path} musi zawierać co najmniej {minimum} znak(ów)',
        inclusive_with_path_one: '{path} musi zawierać co najmniej {minimum} znak',
        inclusive_with_path_other: '{path} musi zawierać co najmniej {minimum} znaki',
        not_inclusive: 'Ciąg musi zawierać więcej niż {minimum} znak(ów)',
        not_inclusive_one: 'Ciąg musi zawierać więcej niż {minimum} znak',
        not_inclusive_other: 'Ciąg musi zawierać więcej niż {minimum} znaki',
        not_inclusive_with_path: '{path} musi zawierać więcej niż {minimum} znak(ów)',
        not_inclusive_with_path_one: '{path} musi zawierać więcej niż {minimum} znak',
        not_inclusive_with_path_other: '{path} musi zawierać więcej niż {minimum} znaki',
      },
      number: {
        inclusive: 'Liczba musi być większa lub równa {minimum}',
        inclusive_with_path: '{path} musi być większe lub równe {minimum}',
        not_inclusive: 'Liczba musi być większa niż {minimum}',
        not_inclusive_with_path: '{path} musi być większe niż {minimum}',
      },
      set: {
        inclusive: 'Nieprawidłowe dane',
        not_inclusive: 'Nieprawidłowe dane',
      },
      date: {
        inclusive: 'Data musi być późniejsza lub równa {- minimum, datetime}',
        not_inclusive: 'Data musi być późniejsza niż {- minimum, datetime}',
      },
    },
    too_big: {
      array: {
        inclusive: 'Tablica może zawierać maksymalnie {maximum} element(ów)',
        inclusive_one: 'Tablica może zawierać maksymalnie {maximum} element',
        inclusive_other: 'Tablica może zawierać maksymalnie {maximum} elementy',
        not_inclusive: 'Tablica może zawierać mniej niż {maximum} element(ów)',
        not_inclusive_one: 'Tablica może zawierać mniej niż {maximum} element',
        not_inclusive_other: 'Tablica może zawierać mniej niż {maximum} elementy',
      },
      string: {
        inclusive: 'Ciąg może zawierać maksymalnie {maximum} znak(ów)',
        inclusive_one: 'Ciąg może zawierać maksymalnie {maximum} znak',
        inclusive_other: 'Ciąg może zawierać maksymalnie {maximum} znaki',
        inclusive_with_path: '{path} może zawierać maksymalnie {maximum} znak(ów)',
        inclusive_with_path_one: '{path} może zawierać maksymalnie {maximum} znak',
        inclusive_with_path_other: '{path} może zawierać maksymalnie {maximum} znaki',
        not_inclusive: 'Ciąg może zawierać mniej niż {maximum} znak(ów)',
        not_inclusive_one: 'Ciąg może zawierać mniej niż {maximum} znak',
        not_inclusive_other: 'Ciąg może zawierać mniej niż {maximum} znaki',
        not_inclusive_with_path: '{path} może zawierać mniej niż {maximum} znak(ów)',
        not_inclusive_with_path_one: '{path} może zawierać mniej niż {maximum} znak',
        not_inclusive_with_path_other: '{path} może zawierać mniej niż {maximum} znaki',
      },
      number: {
        inclusive: 'Liczba musi być mniejsza lub równa {maximum}',
        inclusive_with_path: '{path} musi być mniejsze lub równe {maximum}',
        not_inclusive: 'Liczba musi być mniejsza niż {maximum}',
        not_inclusive_with_path: '{path} musi być mniejsze niż {maximum}',
      },
      set: {
        inclusive: 'Nieprawidłowe dane',
        not_inclusive: 'Nieprawidłowe dane',
      },
      date: {
        inclusive: 'Data musi być wcześniejsza lub równa {- maximum, datetime}',
        not_inclusive: 'Data musi być wcześniejsza niż {- maximum, datetime}',
      },
    },
  },
  storybook: {
    basic: {
      form: {
        textInput: {
          default: {
            label: 'Pole Tekstowe',
            placeholder: 'Podaj wartość',
          },
          disabled: {
            label: 'Zablokowane Pole',
            placeholder: 'Zablokowane',
          },
          invalid: {
            label: 'Nieprawidłowe Pole',
            value: 'Nieprawidłowa wartość',
            error: 'Komunikat błędu',
          },
        },
        textArea: 'Wielowierszowe Pole Tekstowe',
        numberInput: 'Pole Numeryczne',
        checkbox: {
          default: {
            label: 'Pole Wyboru',
          },
          disabled: {
            label: 'Zablokowane Pole Wyboru',
          },
          invalid: {
            label: 'Nieprawidłowe Pole Wyboru',
            error: 'Komunikat błędu',
          },
        },
        radio: {
          default: {
            label: 'Pole Wyboru',
          },
          disabled: {
            label: 'Zablokowane Pole Wyboru',
          },
          invalid: {
            label: 'Nieprawidłowe Pole Wyboru',
            error: 'Komunikat błędu',
          },
        },
        select: {
          label: 'Pole Wyboru',
          placeholder: 'Wybierz wartość',
          option: 'Opcja',
        },
      },
      button: {
        default: 'Przycisk',
        disabled: 'Zablokowany',
        secondary: 'Drugorzędny',
      },
      accordion: {
        heading: 'Nagłówek',
        content: 'Treść',
      },
      dialog: 'Okno Dialogowe',
    },
  },
} satisfies typeof en;

export { pl };
