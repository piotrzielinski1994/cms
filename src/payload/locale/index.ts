import { clientEnv } from '@/env.client';
import {
  DefaultTranslationKeys,
  NestedKeysStripped,
  SupportedLanguages,
  TFunction,
} from '@payloadcms/translations';
import { en } from '@payloadcms/translations/languages/en';
import { pl } from '@payloadcms/translations/languages/pl';
import { pick } from 'ramda';
import { en as customEn } from './en';
import { pl as customPl } from './pl';

// Types ====================================
export type AdminTranslations = TFunction<
  NestedKeysStripped<typeof customEn> | DefaultTranslationKeys
>;

// Variables ====================================
export const adminLocale = {
  list: pick(clientEnv.feature.locale.admin.list, { en, pl } as SupportedLanguages),
  customList: {
    en: customEn,
    pl: customPl,
  },
  default: clientEnv.feature.locale.admin.default,
};

export const contentLocale = {
  list: clientEnv.feature.locale.content.list,
  default: clientEnv.feature.locale.content.default,
};
