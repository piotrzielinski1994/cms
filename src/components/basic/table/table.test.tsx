import { en } from '@/payload/locale/en';
import { render } from '@testing-library/react';
import { ComponentProps } from 'react';
import { describe, expect, it } from 'vitest';
import { axe } from 'vitest-axe';
import { Table } from './table';

const t = en.storybook.basic.table;

describe('Table', () => {
  const [cols, rows] = [3, 5];
  const defaultProps = {
    header: Array.from({ length: cols }, (_, index) => `${t.header} ${index + 1}`),
    footer: Array.from({ length: cols }, (_, index) => `${t.footer} ${index + 1}`),
    body: Array.from({ length: rows }, (_, rowIndex) =>
      Array.from({ length: cols }, (_, index) => `${t.cell} ${rowIndex * cols + index + 1}`),
    ),
  } satisfies ComponentProps<typeof Table>;

  it('should have no accessibility violations', async () => {
    const { container } = render(<Table {...defaultProps} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('should match the snapshot', async () => {
    const { container } = render(<Table {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
