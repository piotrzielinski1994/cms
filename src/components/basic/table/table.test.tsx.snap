// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Table > should match the snapshot 1`] = `
<div>
  <div
    class="w-full overflow-x-auto"
  >
    <table
      class="w-full"
    >
      <thead
        class="font-semibold"
      >
        <tr
          class="bg-primary text-primary-foreground"
        >
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Header 1
          </td>
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Header 2
          </td>
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Header 3
          </td>
        </tr>
      </thead>
      <tbody>
        <tr
          class="bg-background"
        >
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Cell 1
          </td>
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Cell 2
          </td>
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Cell 3
          </td>
        </tr>
        <tr
          class="bg-background1"
        >
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Cell 4
          </td>
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Cell 5
          </td>
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Cell 6
          </td>
        </tr>
        <tr
          class="bg-background"
        >
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Cell 7
          </td>
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Cell 8
          </td>
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Cell 9
          </td>
        </tr>
        <tr
          class="bg-background1"
        >
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Cell 10
          </td>
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Cell 11
          </td>
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Cell 12
          </td>
        </tr>
        <tr
          class="bg-background"
        >
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Cell 13
          </td>
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Cell 14
          </td>
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Cell 15
          </td>
        </tr>
      </tbody>
      <tfoot
        class="font-semibold"
      >
        <tr
          class="bg-background1"
        >
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Footer 1
          </td>
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Footer 2
          </td>
          <td
            class="px-4 py-2 md:px-6 md:py-4"
          >
            Footer 3
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;
